src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+          typename CONSTITUTIVE_TYPE,
+          typename FE_TYPE >
+class SinglePhase :
+  public finiteElement::ImplicitKernelBase< SUBREGION_TYPE,
+                                            CONSTITUTIVE_TYPE,
+                                            FE_TYPE,
+                                            3,
+                                            3 >
+{
+public:
+  /// Alias for the base class;
+  using Base = finiteElement::ImplicitKernelBase< SUBREGION_TYPE,
+                                                  CONSTITUTIVE_TYPE,
+                                                  FE_TYPE,
+                                                  3,
+                                                  3 >;
+
+  /// Number of nodes per element...which is equal to the
+  /// numTestSupportPointPerElem and numTrialSupportPointPerElem by definition.
+  static constexpr int numNodesPerElem = Base::numTestSupportPointsPerElem;
+  using Base::numDofPerTestSupportPoint;
+  using Base::numDofPerTrialSupportPoint;
+  using Base::m_dofNumber;
+  using Base::m_dofRankOffset;
+  using Base::m_matrix;
+  using Base::m_rhs;
+  using Base::m_elemsToNodes;
+  using Base::m_constitutiveUpdate;
+  using Base::m_finiteElementSpace;
+
+
+  /**
+   * @brief Constructor
+   * @copydoc geosx::finiteElement::ImplicitKernelBase::ImplicitKernelBase
+   * @param inputGravityVector The gravity vector.
+   */
+  SinglePhase( NodeManager const & nodeManager,
+               EdgeManager const & edgeManager,
+               FaceManager const & faceManager,
+               localIndex const targetRegionIndex,
+               SUBREGION_TYPE const & elementSubRegion,
+               FE_TYPE const & finiteElementSpace,
+               CONSTITUTIVE_TYPE & inputConstitutiveType,
+               arrayView1d< globalIndex const > const & inputDispDofNumber,
+               string const & inputFlowDofKey,
+               globalIndex const rankOffset,
+               CRSMatrixView< real64, globalIndex const > const & inputMatrix,
+               arrayView1d< real64 > const & inputRhs,
+               real64 const (&inputGravityVector)[3],
+               arrayView1d< string const > const & fluidModelNames ):
+    Base( nodeManager,
+          edgeManager,
+          faceManager,
+          targetRegionIndex,
+          elementSubRegion,
+          finiteElementSpace,
+          inputConstitutiveType,
+          inputDispDofNumber,
+          rankOffset,
+          inputMatrix,
+          inputRhs ),
+    m_X( nodeManager.referencePosition()),
+    m_disp( nodeManager.totalDisplacement()),
+    m_uhat( nodeManager.incrementalDisplacement()),
+    m_gravityVector{ inputGravityVector[0], inputGravityVector[1], inputGravityVector[2] },
+    m_gravityAcceleration( sqrt( inputGravityVector[0] * inputGravityVector[0] +
+                                 inputGravityVector[1] * inputGravityVector[1] +
+                                 inputGravityVector[2] * inputGravityVector[2] ) ),
```

Comment:
Lvarray norm?

Reply:
Done.

Reply:
you could use `LvArray::tensorOps::l2Norm<3>(inputGravityVector);`